src/main/java/uk/gov/hmcts/reform/sscs/service/exceptions/FeignClientErrorDecoder.java

```diff
@@ -0,0 +1,103 @@
+package uk.gov.hmcts.reform.sscs.service.exceptions;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import feign.Request;
+import feign.Request.HttpMethod;
+import feign.Response;
+import feign.codec.ErrorDecoder;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.server.ResponseStatusException;
+import uk.gov.hmcts.reform.sscs.model.HmcFailureMessage;
+import uk.gov.hmcts.reform.sscs.model.single.hearing.HearingRequestPayload;
+import uk.gov.hmcts.reform.sscs.service.AppInsightsService;
+
+import java.nio.charset.StandardCharsets;
+import java.time.LocalDateTime;
+
+@Slf4j
+public class FeignClientErrorDecoder implements ErrorDecoder {
```

Comment:
Can you improve the coverage of your tests:
https://sonarcloud.io/component_measures?id=SSCSHA&metric=new_coverage&pullRequest=72&view=list

Reply:
Test coverage is over 80% at the moment, the only untested parts are the JsonProcessingException's. There are some examples on how to test that here to save time, sscs-hearings-api\src\test\java\uk\gov\hmcts\reform\sscs\service\AppInsightsServiceTest.java

You will need to merge master into your code to get the recent stuff as AppInsightsService has slightly changed